mono/metadata/object.c

```diff
@@ -2834,19 +2852,23 @@ mono_upgrade_remote_class (MonoDomain *domain, MonoObjectHandle proxy_object, Mo
  * the instance of a callvirt of \p method.
  */
 MonoMethod*
-mono_object_get_virtual_method (MonoObject *obj_raw, MonoMethod *method)
+mono_object_get_virtual_method_internal (MonoObject *obj_raw, MonoMethod *method)
 {
 	HANDLE_FUNCTION_ENTER ();
 	MonoMethod *result;
-	MONO_ENTER_GC_UNSAFE;
 	ERROR_DECL (error);
 	MONO_HANDLE_DCL (MonoObject, obj);
 	result = mono_object_handle_get_virtual_method (obj, method, error);
 	mono_error_assert_ok (error);
-	MONO_EXIT_GC_UNSAFE;
```

Comment:
Perhaps this PR should not have any GC mode change changes, i.e. do them in a separate PR.